### ISSUE TITLE
Add support for defining Named Ranges

### ISSUE THREAD

**Issue by p-doyle** [burnash/gspread#833]
**Is your feature request related to a problem? Please describe.**
Unable to define named ranges easily using gspread

**Describe the solution you'd like**
The sheets API provides a method for creating, deleting and updating named ranges that seems easily implementable. 
 https://developers.google.com/sheets/api/samples/ranges

**Describe alternatives you've considered**
You can do this manually by building the request body and calling `Spreadsheet.batch_update()` but it would be nice to have this functionality in the library itself.

**Additional context**
I'm a terrible person and worked on a PR for this before reading the contribution guidelines so I have it standing by in case this is approved.

**Comment by lavigne958** [comment]
Hi @p-doyle,

Thank you for this feature request.

I found your commit about add/update/delete q named range. This is a great proposal.

I did not see your PR though 🙄.

if you are still interested, feel free to open a new one I'll look at it.

From your commit the only missing part I noticed is: add a small test that adds, update then delete a named range. 😉

If you're not interested, let me known and I will chery pick your contribution.
